sycl/doc/extensions/experimental/sycl_ext_codeplay_cuda_tensor_map.asciidoc

```diff
@@ -0,0 +1,370 @@
+= sycl_ext_codeplay_cuda_tensor_map
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+:cuda-guide-using-tma: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#using-tma-to-transfer-multi-dimensional-arrays
+:cuda-guide-async-copies: https://docs.nvidia.com/cuda/cuda-c-programming-guide/#asynchronous-data-copies-using-tensor-memory-access-tma
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) Codeplay Software Limited.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 9 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This is a private extension intended for implementers only. This is an
+experimental extension specification, intended only gather feedback from {dpcpp}
+and {dpcpp}-adjacent implementers.  Interfaces defined in this specification are
+implemented in {dpcpp}, but they are not finalized and may change incompatibly
+in future versions of {dpcpp} without prior notice. *Non-OneAPI implementers
+should not rely on APIs defined in this specification.* It is likely to be
+generalized and significantly change in later revisions as more backend vendors
+implement analogous features of more or less expressivity and generality than
+shown here.
```

Comment:
I'm not wild about adding this new category of "oneapi-only" extension.  I'm guessing this is for XeTLA?  If that is the only library we expect to use this, why not add the support directly in that library, rather than adding a SYCL API for it?  I imagine this would end up calling CUDA APIs or inline asm statements, but I think XeTLA does this already for other devices.  I'd feel differently if we were adding a general API  that other applications could make use of, but that's not what we're doing in this PR.

Reply:
For CUTLASS, actually - but yeah it's not great to be so limiting.

We discussed a couple of ways to do this (with one of my suggestions being interop), but it seems there's little appetite to be including CUDA specific headers in these ports.

To be clear, there's no reason we really *need* to be so limiting with our wording here, I just added it since I'm not completely confident it has uses outside of the CUTLASS case and wanted to limit maintenance burden.

Would relaxing the language here be appropriate?

Reply:
I've used the standard boilerplate language for the `Status` section